avocado/plugins/spawners/process.py

```diff
@@ -76,6 +76,14 @@ async def wait_task(runtime_task):  # pylint: disable=W0221
     @staticmethod
     async def terminate_task(runtime_task):  # pylint: disable=W0221
         runtime_task.spawner_handle.process.terminate()
+        returncode = None
+        try:
+            returncode = await asyncio.wait_for(
+                runtime_task.spawner_handle.process.wait(), 1
```

Comment:
It's very hard to come up with adequate timeouts.  Maybe we'll need to increase it, or maybe make it configurable.  For instance, if we document that the `tearDown` method can take up to 10 seconds, then we'd reconsider this value.


If we have enough hard data to increase this, I'm in favor of doing so.  If not, I'd suggest leaving a timeout that is "large" enough only to allow for the process to respond to the signal.

Reply:
Ok, this is fine for me, let's leave it like it is for now. And maybe in future we can reuse the behavior from legacy runner.

Reply:
Hi @clebergnu, I know that we have already talked about it, but during working on tearDown for avocado-instrumented I become more persuaded that we need to make it configurable. Do you think that it is possible to introduce new spawner config value to increase this wait?

Reply:
> Hi @clebergnu, I know that we have already talked about it, but during working on tearDown for avocado-instrumented I become more persuaded that we need to make it configurable. Do you think that it is possible to introduce new spawner config value to increase this wait?

Yes, I'm about to publish a version with, not one, but two configuration options.

Reply:
> Hi @clebergnu, I know that we have already talked about it, but during working on tearDown for avocado-instrumented I become more persuaded that we need to make it configurable. Do you think that it is possible to introduce new spawner config value to increase this wait?

Hi @richtja ,

Please find this item here https://github.com/avocado-framework/avocado/pull/5788/commits/a0388203f4aa7f0cca8b603e6c4431fad9d322a6 .

The added bonus is that now the Podman spawner will also send `SIGTERM` to the task, so the `tearDown` will also be run under the Podman spawner.  I've tested it and it works!

I've not added tests for these intervals, because I felt like the test you're working on may be suitable here.  Maybe tweaking the soft interval and adding a sleep in the `tearDown` will show that it's working as intended.  But, let's keep in mind that time sensitive tests can be flaky depending on where they run.